dataland-frontend/src/components/general/GenericDataTableModalLink.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <a v-if="validatedData()" @click="openModal()" class="link"
+    >Show {{ linkLabel }}
+    <em class="material-icons" aria-hidden="true" title=""> dataset </em>
+  </a>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import DetailsCompanyDataTable from "@/components/general/DetailsCompanyDataTable.vue";
+import AlignedActivitiesDataTable from "@/components/general/AlignedActivitiesDataTable.vue";
+import NonAlignedActivitiesDataTable from "@/components/general/NonAlignedActivitiesDataTable.vue";
+
+export type GenericsDataTableRequiredData =
+  | {
+      dataId: string;
+      kpiLabel: string;
+      kpiKey: string;
+      content: { [dataId: string]: Array<object | string> };
+      columnHeaders: Record<string, unknown>;
+    }
+  | undefined;
+
+export default defineComponent({
+  name: "GenericDataTableModalLink",
+  components: { DetailsCompanyDataTable, AlignedActivitiesDataTable, NonAlignedActivitiesDataTable },
+  props: {
+    component: {
+      type: Object as () =>
+        | typeof DetailsCompanyDataTable
+        | typeof AlignedActivitiesDataTable
+        | typeof NonAlignedActivitiesDataTable,
+    },
+    data: {
```

Comment:
If we make this required, there is no need for further non-null/undefined checks everywhere else.

Reply:
done